packages/router/src/router.js

```diff
@@ -38,10 +38,26 @@ export function RouterProvider( { children } ) {
 	);
 
 	useEffect( () => {
+		// `/wp_template_part/all` path is no longer used and redirects to
+		// Patterns page for backward compatibility.
+		const { path } = location?.params;
+
+		if ( path === '/wp_template_part/all' ) {
+			history.push( {
+				path: '/patterns',
+			} );
+			setLocation( {
+				...location,
+				params: {
+					path: '/patterns',
+				},
+			} );
+		}
+
 		return history.listen( ( { location: updatedLocation } ) => {
 			setLocation( getLocationWithParams( updatedLocation ) );
 		} );
-	}, [] );
+	}, [ location ] );
```

Comment:
This is not good, this would cause the `history.listen` listener to be re-created on every location change.